crates/sui-framework/packages/dwallet-system/sources/ethereum_state.move

```diff
@@ -164,7 +183,11 @@ module dwallet_system::ethereum_state {
         finality_update: vector<u8>,
         optimistic_update: vector<u8>,
         eth_state: vector<u8>,
-    ): (vector<u8>, u64, vector<u8>);
+        beacon_block: vector<u8>,
+        beacon_block_body: vector<u8>,
+        beacon_block_execution_payload: vector<u8>,
+        beacon_block_type: vector<u8>,
+    ): (vector<u8>, u64, vector<u8>, vector<u8>);
```

Comment:
It's hard to understand what this return value is, maybe replace it with a struct? IIRC there's a straight forward way to return structs from native functions

Reply:
I wish, but I couldn't find a way to do it.
Do you know how?

Reply:
Take a look [here](https://github.com/dwallet-labs/dwallet-network/blob/faff61ba3bf8a61d970e91f4cd79249d2d41349a/crates/pera-framework/build/Pera/sources/groth16.move#L89)

Reply:
Take a look [here](https://github.com/dwallet-labs/dwallet-network/blob/29929ded135f05578b6ce33b52e6ff5e894d0487/crates/pera-framework/packages/pera-framework/sources/crypto/groth16.move#L84)

Reply:
Not really but you can take a look at [this one](https://github.com/dwallet-labs/dwallet-network/blob/29929ded135f05578b6ce33b52e6ff5e894d0487/crates/pera-framework/packages/pera-framework/sources/crypto/groth16.move#L89), do it only if it's easy